### PR TITLE
ENH: expose make_norm_from_scale

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

Internally we now make most norms from scales using `colors._make_norm_from_scale`.  This simply hoists that to public API so downstream libraries can use it as well...

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: doc/api/next_api_changes/behavior/20405-JMK.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Make norm from scale now public
+===============================
+
+Downstream libraries can take advantage of ``make_norm_from_scale`` in 
+``colors.py`` to create a `~.colors.Normalize` subclass directly from
```
Comment (QuLogic) [inline doc/api/next_api_changes/behavior/20405-JMK.rst:5]: Probably should just make this a full-on reference?

Comment (jklymak) [reply doc/api/next_api_changes/behavior/20405-JMK.rst:5]: Yeah, for some reason that wasn't working...

Comment (QuLogic) [reply doc/api/next_api_changes/behavior/20405-JMK.rst]: Ah, an entry needs to be added to `doc/api/colors_api.rst`.

**Review by QuLogic — APPROVED** [review]

File: doc/api/next_api_changes/behavior/20405-JMK.rst (hunk 1 above)
Comment (anntzer) [reply doc/api/next_api_changes/behavior/20405-JMK.rst:5]: I think this still needs to be changed to e.g.
```rst
take advantage of `.colors.make_norm_from_scale`
```
?

File: doc/api/next_api_changes/behavior/20405-JMK.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+Make norm from scale now public
+===============================
+
+Downstream libraries can take advantage of ``make_norm_from_scale`` in 
+``colors.py`` to create a `~.colors.Normalize` subclass directly from 
+an existing scale.  Usually norms have a scale, and the advantage of 
+having a  `~.scale.ScaleBase` attached to a norm is to provide a 
+default scale for the colorbar.
```
Comment (anntzer) [inline doc/api/next_api_changes/behavior/20405-JMK.rst:8]: perhaps be more explicit: "provide a scale, and associated tick locators and formatters, for the colorbar"

**Review by anntzer — APPROVED** [review]
Just one minor nit re docs; can self-merge after addressing it.

**Pull request merged**